### DISCUSSION TITLE
3.0.0 Release, and moving the project to Maintenance Mode.

### DISCUSSION THREAD

**Discussion by AdamBebko** [BioMotionLab/TUX#62] [General]
Hello all, I'm excited to announce the 3.0.0 release of bmlTUX with several improvements and features.

The most exciting feature is continuous data recording and playback. This is a feature that took a long time to get right, and is battle-tested in our own experiments. It is fairly optimized and extendable. I hope you find it useful. See docs for more info.

Several quality of life improvements, fixes, and documentation improvements. Big thanks to DerMilchmann and A-Ivan for their contributions to this.

The most exciting (but also most breaking) change here is that everything that you will need now resides in the same namespace `bmlTUX`.  This will simplify the process for new users, although you'll have to adjust your existing code.

I have also tested bmlTUX with the Unity 2021 tech stream, and it seems to work flawlessly. The hours and hours spent on reorderable lists in the design files are now built into Unity! Too bad they didn't do that 2 years ago :(.

Although this is an exciting moment for bmlTUX, it also comes with some sad news. I have recently left BioMotionLab and have started a new job. Since I'm not funded anymore for maintaining this project, I won't be able to work on any major releases. I will still be able to monitor the project for bugs, and provide some help for smaller things, but I don't plan on making any new features. If you are using bmlTUX and have a request for something a bit more major please contact myself or Nikolaus Troje. Perhaps we can figure something out.

I will still be accepting community contributions. If you think you can improve bmlTUX, please let me know what you're planning. If I think it makes sense to tackle, you can make a fork and submit a pull request. I'd be happy to go over it! 

All the best and happy coding,
Adam

**Comment by DerMilchmann** [comment]
Thank you for your great work!

**Comment by A-Ivan** [comment]
This is great, thanks Adam.
Good luck at your new job.
